certbot/certbot/display/obj.py

```diff
@@ -0,0 +1,530 @@
+"""This modules define the actual display implementations used in Certbot"""
```

Comment:
So it means that ultimately (once the imports done for retrocompatiblity in `display.util` are removed), `FileDisplay` and `NonInteractiveDisplay` classes are removed from the public API?

Reply:
Yeah that's what I was thinking. With the exception of `HelpfulArgumentParser` which is also internal and can continue to use the classes, I feel like these classes are now implementation details of `display.util` and it doesn't feel necessary to me to have them in the public API. What do you think?

Reply:
Well, in this case it seems strange that setting the concrete display with `set_display` is part of the public API, but we do not provide any public concrete implementation of the interface accepted by this setter.

If we go that way, I think we should either provide factories to generate them, like `generate_display(noninteractive=False)`, returning the interface while making the actual instantiation internally. Or we could also make the setter internal and configure the module only through public method, making the display effectively internal (and the public interface useless by the way).

Reply:
Thanks for pointing out that `set_display` is public. I missed that and I agree that's awkward with private display implementations.

What if we only put functions somewhere in `certbot._internal.display` to get and set the display? The message in the error raised by `get_display` and therefore the `display.util` functions if a display isn't set is just something like:

> This function was called too early in Certbot's execution as the display utility hasn't been configured yet.

I doubt it really matters much, but we could also further specify when the module will be ready by saying something like it's done in `certbot.main.main` after argument parsing.

While I think this design of a utility that gets set up automagically by Certbot internals sometime during startup is a little awkward, I think it buys us the most flexibility. It would mean that users couldn't set or define their own display implementations, but I've never seen anyone do that. https://github.com/certbot/certbot/issues/8434 also exists because Erica and Alex wanted to expand the API of IDisplay but didn't do so because it's technically a breaking change if anyone else was implementing that interface. This would also mean that we wouldn't add a (public) replacement for `IDisplay` as part of ripping out `zope.interface`.

Thoughts?

Reply:
To be clear, I think I've seen 3rd party plugins use the `IDisplay` interface which they could still do with this plan through `display.util`. They just couldn't set or implement their own.

Reply:
Ok, the plan sounds good to me, if plugins actually do not set themselve the display service.

Reply:
Done.